vm-virtio/src/queue.rs

```diff
@@ -155,12 +157,61 @@ impl<'a> DescriptorChain<'a> {
         }
     }
 
+    pub fn new_from_indirect(&self) -> Option<DescriptorChain> {
+        if !self.is_indirect() {
+            return None;
+        }
+
+        let desc_head = self.addr;
+        self.mem.checked_offset(desc_head, 16)?;
+
+        // These reads can't fail unless Guest memory is hopelessly broken.
+        let desc = match self.mem.read_obj::<Descriptor>(desc_head) {
+            Ok(ret) => ret,
+            Err(_) => {
+                // TODO log address
```

Comment:
Fix the TODO or drop it?

Reply:
That's a copy/paste from `checked_new`. If we want to get rid of this one, we should probably get rid of both. Perhaps on a separate patch.